src/main/java/hudson/remoting/Engine.java

```diff
@@ -196,11 +203,87 @@ public void removeListener(EngineListener el) {
         events.remove(el);
     }
 
-    @SuppressWarnings({"ThrowableInstanceNeverThrown"})
     @Override
     public void run() {
+        try {
+            IOHub hub = IOHub.create(executor);
+            try {
+                SSLContext context;
+                // prepare our SSLContext
+                try {
+                    context = SSLContext.getInstance("TLS");
+                } catch (NoSuchAlgorithmException e) {
+                    throw new IllegalStateException("Java runtime specification requires support for TLS algorithm", e);
+                }
+                char[] password = "password".toCharArray();
+                KeyStore store;
+                try {
+                    store = KeyStore.getInstance("JKS");
+                } catch (KeyStoreException e) {
+                    throw new IllegalStateException("Java runtime specification requires support for JKS key store", e);
+                }
+                try {
+                    store.load(null, password);
+                } catch (NoSuchAlgorithmException e) {
+                    throw new IllegalStateException("Java runtime specification requires support for JKS key store", e);
+                } catch (CertificateException e) {
+                    throw new IllegalStateException("Empty keystore", e);
+                }
+                KeyManagerFactory kmf;
+                try {
+                    kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+                } catch (NoSuchAlgorithmException e) {
+                    throw new IllegalStateException("Java runtime specification requires support for default key manager", e);
+                }
+                try {
+                    kmf.init(store, password);
+                } catch (KeyStoreException e) {
```

Comment:
These exceptions **will never happen** on a valid JVM... if they happen you are not running on a JVM